cirq-web/cirq_web/bloch_sphere/bloch_sphere.py

```diff
@@ -0,0 +1,140 @@
+
+    def _repr_html_(self):
+        """Allows the object's html to be easily displayed in a notebook
+        by using the display() method.
+
+        If display() is called from the command line, [INSERT HERE]
+        """
+
+        bundle_script = super().get_bundle_script()
+        return f"""
+        <meta charset="UTF-8">
+        <div id="container"></div>
+        {bundle_script}
+        <script>
+        CirqTS.showSphere('{self.sphere_json}', '{self.vector_json}');
+        </script>
+        """
+
+    def generate_HTML_file(
+        self, output_directory='./', file_name='bloch_sphere.html', open_in_browser=False
+    ):
+        """Generates a portable HTML file of the bloch sphere that
+        can be run anywhere. Prints out the absolute path of the file to the console.
+
+        Args:
+            output_directory: the directory in which the output file will be
+            generated. The default is the current directory ('./')
+
+            file_name: the name of the output file. Default is 'bloch_sphere'
+
+            open: if True, opens the newly generated file automatically in the browser.
+
+        Returns the path of the HTML file.
+
+        For now, if ran in a notebook, this function just returns. Support for downloading
+        the HTML file via the browser can be added later.
+        """
+
+        template_div = f"""
+        <meta charset="UTF-8">
+        <div id="container"></div>
+        """
+
+        template_script = f"""
+        <script>
+        CirqTS.showSphere('{self.sphere_json}', '{self.vector_json}');
+        </script>
+        """
+
+        bundle_script = bundle_script = super().get_bundle_script()
+        contents = template_div + bundle_script + template_script
+        path_of_html_file = widget.write_output_file(output_directory, file_name, contents)
+
+        if open_in_browser:
+            webbrowser.open(str(path_of_html_file), new=2)  # 2 opens in a new tab if possible
+
+        return path_of_html_file
+
+    def _convertSphereInput(self, radius):
+        if radius <= 0:
+            raise (BaseException('You must input a positive radius for the sphere'))
+
+        obj = {'radius': radius}
+        return json.dumps(obj)
+
+    def _createVector(self, state_vector):
+        """Any state_vector input will need to come from cirq.to_valid_state_vector,
+        so we can assume that a valid state_vector will be passed in.
+        """
+        bloch_vector = bloch_vector_from_state_vector(state_vector, 0)
+
+        return bloch_vector
+
+    def _createRandomVector(self):
+        random_vector = random_superposition(2)
+        state_vector = to_valid_state_vector(random_vector)
+        bloch_vector = bloch_vector_from_state_vector(state_vector, 0)
+        return bloch_vector
+
+    def _serializeVector(self, x, y, z, length=5):
+        # .item() bc input is of type float32, need to convert to seralizable type
```

Comment:
```suggestion
        # .item() because input is of type float32, need to convert to serializable type
```

Also, can we try whether `cirq.to_json()` would work on the vector?